lib/buffer.js

```diff
@@ -58,7 +58,7 @@ createPool();
 
 
 function SlowBuffer(length) {
-  if (length < 0)
+  if (length < 0 || typeof length !== 'number')
```

Comment:
Both expressions are evaluated for positive integers anyway. So there is only upside to checking type first.